axoupdater/src/test/helpers.rs

```diff
@@ -0,0 +1,216 @@
+use std::{
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
+
+use crate::{get_config_path, ReleaseSourceType};
+
+static RECEIPT_TEMPLATE: &str = r#"{"binaries":[BINARIES],"install_prefix":"INSTALL_PREFIX","provider":{"source":"cargo-dist","version":"0.10.0-prerelease.1"},"source":{"app_name":"APP_NAME","name":"PACKAGE","owner":"OWNER","release_type":"RELEASE_TYPE"},"version":"VERSION"}"#;
+
+/// Generates an install receipt given the specified fields and returns it as a string.
+pub fn install_receipt(
+    app_name: &str,
+    package: &str,
+    owner: &str,
+    binaries: &[String],
+    version: &str,
+    prefix: &str,
+    release_type: &ReleaseSourceType,
+) -> String {
+    let binaries = binaries
+        .iter()
+        .map(|name| format!(r#""{name}""#))
+        .collect::<Vec<String>>()
+        .join(", ");
+    RECEIPT_TEMPLATE
+        .replace("BINARIES", &binaries)
+        .replace("PACKAGE", package)
+        .replace("OWNER", owner)
+        .replace("APP_NAME", app_name)
+        .replace("INSTALL_PREFIX", &prefix.replace('\\', "\\\\"))
+        .replace("VERSION", version)
+        .replace("RELEASE_TYPE", &release_type.to_string())
+}
+
+/// Generates an install receipt given the specified fields and writes it to disk at the appropriate location in `config_path`. The path to the new file is returned.
+#[allow(clippy::too_many_arguments)]
+pub fn write_receipt(
```

Comment:
Do these other functions need to be pub?

Reply:
Probably not. I was exposing them for if other apps chose to roll more of their own tests, but we could also declare these private to ourselves.

Reply:
Let's make them private for now

Reply:
Done.